lib/plugins/threaded-signing/plugin.c

```diff
@@ -254,30 +287,34 @@ threaded_openssl_get_signature(sv_threaded_plugin_t *self,
   SignedVideoReturnCode status = SV_OK;
 
   g_mutex_lock(&self->mutex);
-  if (self->plugin_state == THREADED_SIGNING_HAS_SIGNATURE && self->signature_info) {
-    if (self->signature_info->signature_size > max_signature_size) {
+  if (self->output_buffer_idx > 0) {
```

Comment:
I would recommend changing to check if ```output_buffer_idx == 0``` and if so jump straight to line 319 (unlock).
That will avoid nested if:s which makes the code harder to read.